### ISSUE TITLE
Bug: "undefined method `env' for MyProject::Rails:Module (NoMethodError)" upgrading to 3.6.0+

### ISSUE THREAD

**Issue by Zeneixe** [ViewComponent/view_component#1891]
We can't upgrade our Ruby on Rails project from view_component 3.5.0 to 3.6.0 (nor 3.7.0) due to this error appearing while loading the app's code.

To avoid distractions and debugging purposes I've renamed our project to `MyProject`

```
NoMethodError: undefined method `env' for MyProject::Rails:Module (NoMethodError)

    if Rails.env.development? || Rails.env.test?
            ^^^^
/home/circleci/project/vendor/bundle/ruby/3.2.0/gems/view_component-3.7.0/lib/view_component/base.rb:222:in `<class:Base>'
```

I'm not sure why this is happening on this version specifically but it should be easy enough to fix.

### Steps to reproduce
Upgrade our current `view_component` version from `3.5.0` to `3.6.0 or 3.7.0`

### Suggested fix

Change this line:

https://github.com/ViewComponent/view_component/blob/9f61796a47c9dbddf4c93f9f0c461c776647fa44/lib/view_component/base.rb#L222

Have the Rails module prepended with `::`:

```
    if ::Rails.env.development? || ::Rails.env.test?
```

**Comment by camertron** [comment]
Hey @Zeneixe, sorry to hear you're having trouble upgrading. First off, can you confirm you're trying to upgrade to v3.6.0 or v3.7.0? I don't believe a v0.5.0 exists, and if it did, it would be _very_ old 😅 

It looks like the line you mentioned was introduced in v3.6.0, and there should indeed be a `::` in front of the `Rails` constant (although I'm a little concerned that your app appears to define another constant called `Rails`?) Would you mind submitting a PR?

**Comment by rmacklin** [comment]
> I'm not sure why this is happening on this version specifically but it should be easy enough to fix.

That line was added in https://github.com/ViewComponent/view_component/pull/1841, which was first included in v3.6.0

**Comment by camertron** [comment]
Great, thank you!
